### ISSUE TITLE
Allow non-administrative users the ability to publish/unpublish nodes

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#815]
There has been a long-standing problem with Drupal sites that the publish/unpublish option is tied to the "Administer nodes" permission. In most cases users who should be allowed to publish/unpublish nodes are not also intended to administer all content on the site, but with core alone that is the only option.

Drupal contrib module that solves this problem"
* [Override node options](override_node_options)

**Comment by docwilmot** [comment]
Good idea. Would you propose including all the new permissions the module provides or just publish/unpublish?

**Comment by jromine** [comment]
I agree; one use case I've had is classified ads postings, where we wanted the node author to be able to publish & unpublish her nodes.

**Comment by jenlampton** [comment]
I think all the options would be ideal, but publish/unpublish is the only necessity.

**Comment by klonos** [comment]
Yep, very useful. Should be in core.

**Comment by Dinornis** [comment]
👍 for additional permissions:

- Publish/Unplublish own content
- Publish/Unplublish any content

**Comment by olafgrabienski** [comment]
I agree as well, we should have more fine-grained permissions for content editors in core.

Apart from that I saw a contrib module ported by @jenlampton: https://github.com/backdrop-contrib/override_node_options . Has someone used or tested it yet?

**Comment by jenlampton** [comment]
I'm using it on 2 sites (including backdropcms.org) and it seems to be working pretty well. Would love more testers though!

**Comment by olafgrabienski** [comment]
@jenlampton Thanks, good to know! Will try to remember to share my experience when I use the module (not yet the case).

**Comment by olafgrabienski** [comment]
Oh, I forgot to give feedback! I'm using the *Override Node Options* on at least one site and after an issue with installation (which has been fixed) I don't remember to have encountered any further issues.

Apart of that, is there still the wish to move the functionality into core?

**Comment by docwilmot** [comment]
But this issue is just for publish/unpublish. The ONO module does a lot more. Are we talking about just adding publish options to core and leave the rest in contrib, or move the entire functionality into core? Whats the 80/20 usage likely here?

**Comment by olafgrabienski** [comment]
For reference, these are the options handled by *Override Node Options*:

- published
- promote to front page
- sticky
- revision
- revision log entry
- authored on
- authored by
- URL alias
- comment setting

In my opinion, all the options make sense in certain situations, and it would be consistent to have them all in core for more fine-grained control, because *Adminster content* permission is very wide. On all of my Backdrop and Drupal sites, I have used every option, but not all on one site to be honest.

**Comment by jenlampton** [comment]
My personal usage of ONO module is like this:

Module used: 40% of my sites

- Published overridden: 100% of sites where module is used
- Promoted overridden: 25% of sites where module is used
- Author overridden: 5% of sites where module is used
- Everything else: never used

**Comment by stpaultim** [comment]
If we do this, we might want to include scheduling posts as an option (suggested by @klonos).
https://github.com/backdrop/backdrop-issues/issues/4543
